.github/workflows/test-check-transformers.yaml

```diff
@@ -15,9 +15,41 @@ env:
   CLEARML_API_SECRET_KEY: ${{ secrets.CLEARML_API_SECRET_KEY }}
 
 jobs:
+  detect-changes:
+    runs-on: ubuntu-latest
+
+    outputs:
+      changes-present: ${{ steps.changed-files.outputs.any_modified }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            **
+            !examples/**
+            !tests/e2e/**
+            !tests/lmeval/**
+            !tests/examples/**
+            !**/*.md
+            !.github/**
+            .github/workflows/test-check-transformers.yaml
```

Comment:
Does this mean check every file (**) but exclude `examples/`, ..., exclude `.github/`, but re-include `.github/workflows/test-check-transformers.yaml`? so the order matters?

Reply:
> Does this mean check every file (**) but exclude `examples/`, ..., exclude `.github/`, but re-include `.github/workflows/test-check-transformers.yaml`?

Basically, yes.

> so the order matters?

I _think_ so, based on my reading through the docs. Notably, starting with v13, they switched/supported glob pattern matching:

https://github.com/tj-actions/changed-files/releases/tag/v13

Following the link about the glob pattern matching in those release notes, it seems to suggest order is relevant:

> `!`: At the start of a pattern makes it negate previous positive patterns. It has no special meaning if not the first character.